admin/data/sql/migrations/20190918140848.do.pupil-state.sql

```diff
@@ -0,0 +1,21 @@
+alter table mtc_admin.pupil
+	add currentCheckId int
+
+alter table mtc_admin.pupil
+	add checkComplete bit
+	constraint DF_checkComplete_Default default (0)
+
+alter table mtc_admin.pupil
+	add restartActive bit
+	constraint DF_restartActive_Default default (0)
+
+alter table mtc_admin.pupil
+	add attendanceId int
+
+alter table mtc_admin.pupil
+	add constraint pupil_check_id_fk
+		foreign key (currentCheckId) references mtc_admin.[check]
+
+alter table mtc_admin.pupil
+	add constraint pupil_attendanceCode_id_fk
+		foreign key (attendanceId) references mtc_admin.attendanceCode
```

Comment:
I am slightly surprised.  Linking directly to the `attendanceCode` removes the pupil-attendance join table so that it will be quicker.   But don't we need to know any context about this?  At the moment we store the user that made the non-attendance record and the timestamp.  Are these no longer required, or will they be stored elsewhere?

Reply:
see above.  same strategy as restart - pupilAttendance is retained as a history log, this merely serves as a way to query the pupils state without partitioning the attendance table into the register/status page queries.

Reply:
That works for me 😄